cachecontrol.go

```diff
@@ -29,15 +29,15 @@ func ParseCacheControl(input string) (CacheControl, error) {
 		var token string
 		switch input[pos] {
 		case '"':
-			token = readString(input, pos+1, "\"")
+			token = readUntil(input, pos+1, "\"")
 			pos += len(token) + 1
 		case ',', '\n', '\r', ' ', '\t':
 			continue
 		case '=':
 			isValue = true
 			continue
 		default:
-			token = readString(input, pos, "\"\n\t\r ,=")
+			token = readUntil(input, pos, "\"\n\t\r ,=")
```

Comment:
I'm not quite sure what's different about this func, but can't it be simplified to a call to [`strings.IndexAny`](https://godoc.org/strings#IndexAny)? I.e., something like:

```
until := strings.IndexAny(input[pos:], "\"\n\t\r ,=")
// check if until is -1
// use input[pos:pos+until]
```